### ISSUE TITLE
Characters are missing!

### ISSUE THREAD

**Issue by RArunRaja** [ruffle-rs/ruffle#3393]
My SWF files have Icelandic fonts. Some Icelandic characters are missing, mostly on the dynamic texts.

Please check the images!
![Issue](https://user-images.githubusercontent.com/50624978/108831877-bdb09800-75f0-11eb-855b-f2511b6a7bf3.png)

![No issue](https://user-images.githubusercontent.com/50624978/108834981-d622b180-75f4-11eb-81fc-966143d9b2ee.png)

Please check the script here!

```
 <script>
var swfobject = {};

swfobject.embedSWF = function(url, cont, width, height){
    var ruffle = window.RufflePlayer.newest(),
        player = Object.assign(document.getElementById(cont).appendChild(ruffle.createPlayer()), {
            width: width,
            height: height,
            style: 'width: ' + width + 'px; height: ' + height + 'px',
        });
    
    player.load({ url: url });
}

swfobject.embedSWF('index.swf', 'ruffle', 900, 900);
        </script>
     
```

**Comment by ousia** [comment]
@RArunRaja, many thanks for reporting the issue.

You don’t provide an address to access the SWF file (or attach the file to the issue). I’m afraid I can only guess.

If the first image contains what _Ruffle_ displays, I think that your SWF file might include fonts, but the fonts might not embed all characters required to display the text.

I’d be surprised if your issue isn’t a duplicate of #1862 (long story short: _Ruffle_ has problems to display non-ASCII characters when the font in the SWF hasn’t them embedded).

I think that the title of your issue should read “characters are missing” (or explain why fonts are missing).

Could you also provide the address to access and test your SWF file?

If this wasn’t possible, please attach the SWF file here (just renaming the extension from `.swf` to `.zip` [and warning that the file isn’t compressed] does the job).

I hope it helps.

**Comment by RArunRaja** [comment]
@ousia Thank you for reaching out!

http://skolavefurinn.is/sites/default/files/_flash_player/utprentun_forrit/index.swf

Here you can find the sample file.

I really appreciate your help.

**Comment by ousia** [comment]
@RArunRaja, many thanks for providing the address.

Your file uses a sans-serif font (object 149) with no characters embedded. _Ruffle_ has to use a system font, but until #1826 is solved, it uses an embedded font which contains few non-ASCII characters.

Edittext fields use the sans-serif font (object 149). There are three objects for that: 150, 207 and 270.

Button (object 58) contains a text area (object 153) that uses the font (object 149) and an edittext (object 150). This is problematic for `label = "Vídeóleiðbeiningar";` and `label = "Skoða"`. It isn’t noticeable in `label = "Bekkir";` and `label = "Prenta";`.

Check box (object 268) contains a text area (object 153) that uses the font (object 149) and an edittext (object 150). But as far as I know, it isn’t noticeable.

Radio button (object 288) is related to a label (object 271) which uses an edittext (object 270) deploying the non-embedded font (object 149).

Again, issue #1862 is aware of this issue. I’d close this issue and subscribe to issue #1862 (since this one is duplicated).

I hope it helps.

**Comment by RArunRaja** [comment]
So, I think I should wait for the fix via #1862

**Comment by torokati44** [comment]
I think most (all?) characters needed for this piece of content were added by #6224.
Feel free to reopen if that's not the case.
